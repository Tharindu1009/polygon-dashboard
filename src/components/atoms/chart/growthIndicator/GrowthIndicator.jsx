
import './styles/growthIndicator.scss';
import * as React from 'react';
import CN from "classnames";
import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';

function GrowthIndicator({ direction, percentage, content }) {

    const IconClasses = CN({
        "icon icon__up": direction === "up",
        "icon icon__down": direction === "down",
    });

    return (
        <div data-testid="growthIndicator" className='growthIndicator'>
            {direction === "up" && <ArrowUpwardIcon data-testid="upArrow" className={IconClasses} />}
            {direction === "down" && <ArrowDownwardIcon data-testid="downArrow" className={IconClasses} />}
            <div className='content'>
                <div className='percentage'>{percentage}</div> %
                <div className='descrtption'>{content}</div>
            </div>
        </div>
    );
}

export default GrowthIndicator;

