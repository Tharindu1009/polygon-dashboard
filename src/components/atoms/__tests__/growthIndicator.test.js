import { render, screen, cleanup } from "@testing-library/react";
import TestRenderer from 'react-test-renderer';
import GrowthIndicator from "../chart/growthIndicator/GrowthIndicator";

afterEach(()=>{
    cleanup();
})

test('should render growthIndicator component', () => {
    render(<GrowthIndicator direction="up" percentage="1.10" content="Since yesterday" />);
    const growthIndicator = screen.getByTestId('growthIndicator');
    expect(growthIndicator).toBeInTheDocument();
    expect(growthIndicator).toHaveTextContent('1.10 %');
    expect(growthIndicator).toHaveTextContent('Since yesterday');
})

test('should render growthIndicator component with up arrow when direction is up', () => {
    render(<GrowthIndicator direction="up" percentage="1.10" content="Since yesterday" />);
    const upArrow = screen.getByTestId('upArrow');
    expect(upArrow).toBeInTheDocument();
})

test('should render growthIndicator component with down arrow when direction is down', () => {
    render(<GrowthIndicator direction="down" percentage="1.10" content="Since yesterday" />);
    const downArrow = screen.getByTestId('downArrow');
    expect(downArrow).toBeInTheDocument();
})

test('matches growthIndicator snapshot', () => {
    const tree = TestRenderer.create(<GrowthIndicator direction="up" percentage="1.10" content="Since yesterday" />).toJSON();
    expect(tree).toMatchSnapshot();
})