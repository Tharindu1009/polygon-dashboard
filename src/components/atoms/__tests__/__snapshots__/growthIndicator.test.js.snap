// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches GrowthIndicator snapshot 1`] = `
<div
  className="growthIndicator"
  data-testid="growthIndicator"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon icon__up css-i4bv87-MuiSvgIcon-root"
    data-testid="upArrow"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  <div
    className="content"
  >
    <div
      className="percentage"
    >
      1.10
    </div>
     %
    <div
      className="descrtption"
    >
      Since yesterday
    </div>
  </div>
</div>
`;

exports[`matches growthIndicator snapshot 1`] = `
<div
  className="growthIndicator"
  data-testid="growthIndicator"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon icon__up css-i4bv87-MuiSvgIcon-root"
    data-testid="upArrow"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  <div
    className="content"
  >
    <div
      className="percentage"
    >
      1.10
    </div>
     %
    <div
      className="descrtption"
    >
      Since yesterday
    </div>
  </div>
</div>
`;

exports[`matches snapshot 1`] = `
<div
  className="growthIndicator"
  data-testid="growthIndicator"
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon icon__up css-i4bv87-MuiSvgIcon-root"
    data-testid="upArrow"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="m4 12 1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
    />
  </svg>
  <div
    className="content"
  >
    <div
      className="percentage"
    >
      1.10
    </div>
     %
    <div
      className="descrtption"
    >
      Since yesterday
    </div>
  </div>
</div>
`;
